File: WebFt30/src/app/core/store/models/threshold.ts
export enum ThresholdParameter {
  EventLoss = 0,
  TotalLoss = 1,
  EventReflectance = 2,
  SectionAttenuation = 3,
  SectionLoss = 4,
  SectionLengthChange = 5,
  PortHealth = 6
}

export class Threshold {
  id!: number;
  parameter: ThresholdParameter;

  isMinorOn = false;
  minor: number | null = null;
  isMajorOn = false;
  major: number | null = null;
  isCriticalOn = false;
  critical: number | null = null;

  constructor(param: ThresholdParameter) {
    this.parameter = param;
  }

  // enabled/disabled threshold generally
  public isOn(): boolean {
    if (this.isMinorOn || this.isMajorOn || this.isCriticalOn) return true;
    return false;
  }
}

File: Backend/src/Fibertest30.Application/AlarmProfile/Models/Threshold.cs
namespace Fibertest30.Application;

public class Threshold
{
    public int Id { get; init; }
    public ThresholdParameter Parameter { get; init; }

    public bool IsMinorOn { get; set; }
    public double? Minor { get; set; }
    public bool IsMajorOn { get; set; }
    public double? Major { get; set; }
    public bool IsCriticalOn { get; set; }
    public double? Critical { get; set; }


    public Threshold() { }

    public Threshold(ThresholdParameter parameter, double? minor, double? major, double? critical)
    {
        Parameter = parameter;
        Minor = minor;
        Major = major;
        Critical = critical;
    }
}